### PR TITLE
:bug: (RUMM-708): Prevent frequent retry on network layer error

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Increase the delay between batch uploads when an error is detected.

### Motivation

When an app using the SDK is ran behind some firewalls/proxy, the data intake request can fail. This means that the workers are going to retry up to every 2 seconds to send the same batch, leading in more CPU consumption. 

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt [hunk 1]
```diff
@@ -27,20 +28,22 @@ internal class DataUploadRunnable(
...
 
     override fun run() {
         val batch = if (isNetworkAvailable() && isSystemReady()) {
             reader.readNextBatch()
         } else null
+
         if (batch != null) {
             consumeBatch(batch)
         } else {
-            delayTheRunnable()
```
Comment (ncreated) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt:42]: Just confirming. It was:
* `x`, `0.9x`, `0.81x`, ... when the batches were found sequentially (with the `min()` cap),
* jumping discretely to `x = 20s` on the first missing batch.

Now we do it smoothly:
* `x`, `0.9x`, `0.81x` on batch found (with the `min()` cap),
* `1.1x`, `1.21x`, ... on batch miss (with the `max()` cap),

right?

File: dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt [hunk 2]
```diff
@@ -376,43 +377,145 @@ internal class DataUploadRunnableTest {
...
+                assertThat(next)
+                    .isLessThanOrEqualTo(previous)
+                    .isBetween(DataUploadRunnable.MIN_DELAY_MS, DataUploadRunnable.MAX_DELAY_MS)
+                next
+            }
+        }
+    }
+
+    @Test
+    fun `𝕄 reduce delay between runs 𝕎 no batch availabled`(
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt:451]: typo here `available`

Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt:451]: also it should be `M increase the delay`

File: dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt [hunk 3]
```diff
@@ -376,43 +377,145 @@ internal class DataUploadRunnableTest {
...
+                assertThat(next)
+                    .isGreaterThanOrEqualTo(previous)
+                    .isBetween(DataUploadRunnable.MIN_DELAY_MS, DataUploadRunnable.MAX_DELAY_MS)
+                next
+            }
+        }
+    }
+
+    @Test
+    fun `𝕄 reduce delay between runs 𝕎 batch fails and should be retried`(
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnableTest.kt:482]: M increase delay between runs W batch fails and should be retried

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataUploadRunnable.kt:42]: For how it used to be, you're completely right, but on the way it's done now, to be more accurate it is : 
- `𝑥`, `0.9𝑥`, `0.81x`, … when the batch is found and processed successfully, that is when the upload returns 200, 4xx (indicating an invalid format or request) or an unknown error occurs (meaning the batch might be faulty)
- `𝑥`, `1.1𝑥`, `1.21𝑥`, … when there's no batch or the batch process fails, that is when the upload return 5xx, 403 (invalid or expired client token), or a network exception occurs (e.g.: DNS, Connection timeout, …)

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**